compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertCores.cpp

```diff
@@ -30,74 +40,36 @@ void getAttributeMapping(SmallVector<scf::ForallOp> forallOps,
   }
 }
 
-/// Insert workgroups around `scf::Forall` ops with thread mapping.
-/// NOTE: This can be extended to block mapping as well later.
-LogicalResult insertWorkgroupOps(mlir::ModuleOp moduleOp) {
-  IRRewriter rewriter(moduleOp.getContext());
-  moduleOp->walk([&](scf::ForallOp forallOp) {
-    if (forallOp.getNumResults() != 0) {
-      return WalkResult::advance();
-    }
-    if (!forallOp.getMapping().has_value()) {
-      return WalkResult::advance();
-    }
-    SmallVector<Attribute> threadMapping =
-        llvm::to_vector(forallOp.getMapping()->getValue());
-    if (llvm::any_of(threadMapping, [](Attribute map) {
-          return !llvm::isa<mlir::gpu::GPUThreadMappingAttr>(map);
-        })) {
-      return WalkResult::advance();
-    }
-    if (threadMapping.size() != 2) {
-      return WalkResult::advance();
-    }
-    // Create the workgroup region
-    rewriter.setInsertionPoint(forallOp);
-    auto workgroupOp =
-        rewriter.create<AMDAIE::WorkgroupOp>(rewriter.getUnknownLoc());
-    rewriter.moveOpBefore(forallOp, workgroupOp.getControlCode());
-    return WalkResult::advance();
-  });
-  return success();
-}
-
 /// Insert core ops inside innermost forall ops around computational ops and
 /// add synchronization ops along the way to synchronize with surrounding
 /// dma ops.
-LogicalResult insertCoreOpsInWorkgroup(mlir::ModuleOp moduleOp) {
+LogicalResult insertCoreOps(mlir::ModuleOp moduleOp) {
   IRRewriter rewriter(moduleOp.getContext());
-  WalkResult res = moduleOp->walk([&](AMDAIE::WorkgroupOp workgroupOp) {
-    // Find innermost forall loops
-    SmallVector<scf::ForallOp> innermostForallLoops =
-        getInnermostForallLoops(workgroupOp);
```

Comment:
This utility seems not be used in other places, and can be cleaned as well?

Reply:
Thanks for the catch!